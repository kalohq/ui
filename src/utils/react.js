import ReactDOM from 'react-dom';

/**
 * Focus a ref’d react input when it is mounted
 * @param ref
 */
export function focusOnMount(ref) {
  const el = ReactDOM.findDOMNode(ref);

  if (el) {
    // Give focus and put the cursor at the end of
    // the value.
    el.focus();
    el.value = el.value;
  }
}

/**
 * Select the text of a ref’d react input when it is mounted
 * @param ref
 */
export function selectOnMount(ref) {
  if (ref && ref.select) {
    ref.select();
  }
}
