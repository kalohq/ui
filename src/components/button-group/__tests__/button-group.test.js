/* @flow */
/* eslint-env jest */
import * as React from 'react';
import {sheet} from 'emotion';
import serializer from 'jest-glamor-react';
import renderer from 'react-test-renderer';

import ButtonGroup from 'components/button-group';
import Button from 'components/button';

expect.addSnapshotSerializer(serializer(sheet));

describe('ButtonGroup', () => {
  const defaultProps = {};
  const create = (props = {spacing: true}) =>
    renderer
      .create(
        // $FlowFixMe
        <ButtonGroup {...defaultProps} {...props}>
          <Button>Button One</Button>
          <Button>Button Two</Button>
        </ButtonGroup>
      )
      .toJSON();

  test('should render shallow component ok', () => {
    const element = create();
    expect(element).toMatchSnapshot();
  });
});
