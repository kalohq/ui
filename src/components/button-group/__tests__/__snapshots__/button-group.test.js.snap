// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should render shallow component ok 1`] = `
<div
  className="ui-btn-group ui-btn-group--wide emotion-0 emotion-1"
  data-test="ui-button-group"
  style={Object {}}
>
  <button
    className="ui-btn ui-btn--large ui-btn--primary ui-btn--grouped-with-spacing emotion-0 emotion-1"
    data-test="ui-button"
    disabled={false}
    name={undefined}
    onClick={[Function]}
    role="button"
    style={Object {}}
    type={undefined}
  >
    <div
      className="ui-btn__react-placeholder-message"
    >
      One
    </div>
    <div
      className="ui-btn__react-visible-message"
      name="message"
      title="One"
    >
      One
    </div>
    <div
      className="ui-btn__react-success-message"
    />
  </button>
  <button
    className="ui-btn ui-btn--large ui-btn--primary ui-btn--grouped-with-spacing emotion-0 emotion-1"
    data-test="ui-button"
    disabled={false}
    name={undefined}
    onClick={[Function]}
    role="button"
    style={Object {}}
    type={undefined}
  >
    <div
      className="ui-btn__react-placeholder-message"
    >
      Two
    </div>
    <div
      className="ui-btn__react-visible-message"
      name="message"
      title="Two"
    >
      Two
    </div>
    <div
      className="ui-btn__react-success-message"
    />
  </button>
</div>
`;
